lib/commanded/event/handler.ex

```diff
@@ -487,12 +487,13 @@ defmodule Commanded.Event.Handler do
 
             _ ->
               # retry event, failure count is included in context map
-              {:retry, context}
+              {:retry, failure_context}
           end
         end
 
-        defp record_failure(context) do
-          Map.update(context, :failures, 1, fn failures -> failures + 1 end)
+        defp record_failure(failure_context) do
+          context = Map.update(failure_context.context, :failures, 1, fn failures -> failures + 1 end)
+          %FailureContext{failure_context | context: context}
```

Comment:
Annotation: Wrapping the `context` in the `FailureContext` is the main fix.  The other changes were just to make it more clear you need to return the `FailureContext`.